src/data/repositories/DataSetD2Repository.ts

```diff
@@ -314,10 +317,15 @@ export class DataSetD2Repository implements DataSetRepository {
             }
 
             const existingAttributes = existingDataSet?.attributeValues;
+            const sharingData = this.d2ApiSharing.generateSharingData(dataSet);
 
             const result = {
```

Comment:
naming: result -> can we change it to a more descriptive name?

type:  let's avoid creating anonymous objects with an explicit type. When I put it (the first arg of buildDataEntryForm), I get a type error.